### ISSUE TITLE
Support for Thread affinity needed

### ISSUE THREAD

**Issue by mgubisch** [taskflow/taskflow#190]
Hi,
first of all, great project you put off there, I really like it.
Unfortunately I can not use it for my project because 1 important feature is missing.
I need the possibility to pin certain tasks to a specific worker id (specific CPU core would be even better but not mandatory)
In addition it would be great to explicitly initialize worker threads to CPU cores.

similar the way intel is doing in its GTS project:
https://github.com/GameTechDev/GTS-GamesTaskScheduler/blob/master/source/gts/examples/micro_scheduler/1_initialization/include/2_explicit_init.h

As API I can imagine something like
`task.setAffinity(int workerID); `

Would be really great to see this feature coming soon.

**Comment by Mazrog** [comment]
Hi,
Are there any updates on this?

I would like to use this library on some projects where I use OpenGL and GLFW but there are some calls that can only be called from the thread where they were initialized, so I am kind of stuck at the moment.

The ability to put some "affinity flags" (maybe, to add some classification semantic) to tasks, that can refer to some threads (preventing other to steal from them) would be useful.

Thanks

**Comment by Nahor** [comment]
@tsung-wei-huang, I don't think this worker property feature is what was asked in this ticket. The worker property allows to assign a worker to a specific CPU. What was asked is to assign a task to a specific worker.

And better yet, it would be a set of workers. For instance with Vulkan, one needs to create per-thread resources (pools, queues, ...). Using affinity, one would only need to create the resources for a few of the workers (the "Vulkan workers") instead of each and every worker.
